include/forcing/NullForcingProvider.hpp

```diff
@@ -0,0 +1,271 @@
+    }
+
+    virtual std::vector<double> get_values(const CatchmentAggrDataSelector& selector, data_access::ReSampleMethod m) override
+    {
+        throw std::runtime_error("Called get_values function in NullDataProvider");
+    }
+
+
+    /**
+     * Get whether a param's value is an aggregate sum over the entire time step.
+     *
+     * Certain params, like rain fall, are aggregated sums over an entire time step.  Others, such as pressure, are not
+     * such sums and instead something else like an instantaneous reading or an average value over the time step.
+     *
+     * It may be the case that forcing data is needed for some discretization different than the forcing time step.
+     * These values can be calculated (or at least approximated), but doing so requires knowing which values are summed
+     * versus not.
+     *
+     * @param name The name of the forcing param for which the current value is desired.
+     * @return Whether the param's value is an aggregate sum.
+     */
+    //TODO this function doesn't seem being used
+    inline bool is_param_sum_over_time_step(const std::string& name) {
+        if (name == CSDMS_STD_NAME_RAIN_VOLUME_FLUX) {
+            return true;
+        }
+        if (name == CSDMS_STD_NAME_SOLAR_SHORTWAVE) {
+            return true;
+        }
+        if (name == CSDMS_STD_NAME_SOLAR_LONGWAVE) {
+            return true;
+        }
+        if (name == CSDMS_STD_NAME_LIQUID_EQ_PRECIP_RATE) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Get whether a property's per-time-step values are each an aggregate sum over the entire time step.
+     *
+     * Certain properties, like rain fall, are aggregated sums over an entire time step.  Others, such as pressure,
+     * are not such sums and instead something else like an instantaneous reading or an average value.
+     *
+     * It may be the case that forcing data is needed for some discretization different than the forcing time step.
+     * This aspect must be known in such cases to perform the appropriate value interpolation.
+     *
+     * @param name The name of the forcing property for which the current value is desired.
+     * @return Whether the property's value is an aggregate sum.
+     */
+    //TODO this one used in Bmi_Module_Formulation.hpp and Bmi_Multi_Formulation.hpp
+    inline bool is_property_sum_over_time_step(const std::string& name) override {
+        return is_param_sum_over_time_step(name);
+    }
+
+    const std::vector<std::string> &get_available_variable_names() override {
+        return available_forcings;
+    }
+
+    private:
+
+    /**
+     * @brief Checks forcing vector index bounds and adjusts index if out of vector bounds
+     * /// \todo: Bounds checking is based on precipitation vector. Consider potential for vectors of different sizes and indices.
+     */
+    inline void check_forcing_vector_index_bounds()
```

Comment:
Looks like this was dead code in CsvPerFeatureForcingProvider, sorry. It should be removed.

Reply:
Yes. Just looking at it.